### PR TITLE
CHE-1276: Add exception handler during creation machine from snapshot

### PR DESCRIPTION
**Pull request by akorneta**
When docker image is not exist and snapshot is not null then in case of creating   machine instance from this config, docker client will throw an exception, to avoid this situation we should handle this error. It is proposed to create machine instance  form incoming config(recipe).
@skabashnyuk @vparfonov @garagatyi

### CODE REVIEW

**Comment by garagatyi** [comment]
In general I'm OK with the solution, but I think you can improve & simplify it. 
Instead of cloning the whole origin MachineConfig you can clone just origin source. Then if there is an exception and origin source is not null you replace source in already created machine instance and try to start machine creation again. 
I believe that in that case this PR will introduce less changes and logic will be clearer.

Also it looks like MachineRegistry.addMachine() should throw ConflictException in that case. Can you check that?

**Comment by vparfonov** [comment]
ok

**Comment by garagatyi** [comment]
LGTM
@skabashnyuk Please take a look.

File: plugins/plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerConnector.java [hunk 1]
```diff
@@ -1242,21 +1245,27 @@ private void removeImage(final RemoveImageParams params,URI dockerDaemonUri) thr
...
 
     /**
      * Tag the docker image into a repository.
      *
      * @throws IOException
-     *          when a problem occurs with docker api calls
+     *         when a problem occurs with docker api calls
      */
-    public void tag(final TagParams params) throws IOException {
+    public void tag(final TagParams params) throws ImageNotFoundException,
```
Comment (evoevodin) [inline plugins/plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerConnector.java]: Document ImageNotFoundException

**Comment by voievodin** [comment]
ok :+1:

**Pull request merged**